seqr/views/utils/terra_api_utils.py

```diff
@@ -0,0 +1,177 @@
+        """
+        r = self.__get("register")
+        if r.status_code is not 200:
+            raise TerraAPIException(
+                'Error: called Terra API "register" got status: {} with a reason: {}'.format(r.status_code, r.reason))
+        return json.loads(r.text)
+
+    def list_workspaces(self, fields=None):
+        """Get all the workspaces accessible by the logged-in user.
+
+        Args:
+        fields (str): a comma-delimited list of values that limits the
+            response payload to include only those keys and exclude other
+            keys (e.g., to include {"workspace": {"attributes": {...}}},
+            specify "workspace.attributes").
+        """
+        if fields is None:
+            r = self.__get("api/workspaces")
+        else:
+            r = self.__get("api/workspaces", params = {"fields": fields})
+        if r.status_code is not 200:
+            raise TerraAPIException(
+                'Error: called Terra API "api/workspaces" got status: {} with a reason: {}'.format(r.status_code, r.reason))
+        return json.loads(r.text)
+
+    def get_workspace_acl(self, namespace, workspace):
+        """Request FireCloud access control list for workspace.
+        Args:
+            namespace (str): project to which workspace belongs
+            workspace (str): Workspace name
+        Returns:
+            {
+                "user1Email": {
+                  "accessLevel": "string",
+                  "pending": true,
+                  "canShare": true,
+                  "canCompute": true
+                },
+                "user2Email": {
+                  "accessLevel": "string",
+                  "pending": true,
+                  "canShare": true,
+                  "canCompute": true
+                },
+                "user3Email": {
+                  "accessLevel": "string",
+                  "pending": true,
+                  "canShare": true,
+                  "canCompute": true
+                }
+              }
+        """
+        uri = "api/workspaces/{0}/{1}/acl".format(namespace, workspace)
+        r = self.__get(uri)
+        if r.status_code is not 200:
+            raise TerraAPIException(
+                'Error: called Terra API "{}" got status: {} with a reason: {}'.format(uri, r.status_code, r.reason))
+        return json.loads(r.text)['acl']
+
+    def get_staffs(self):
+        """Request members of seqr staff group.
+        """
+        r = self.__get("api/groups/seqr-staffs")
+        if r.status_code is not 200:
+            raise TerraAPIException(
+                'Error: called Terra API "api/groups/seqr-staffs" got status: {} with a reason: {}'.format(r.status_code, r.reason))
+        group = json.loads(r.text)
+        return group["adminsEmails"]
+
+    def is_staff(self, user_email):
+        return user_email in self.get_staffs()
+
+
+service_account_session = AnvilSession(scopes = scopes)
```

Comment:
I'm a bit confused how this works with multiple concurrent users. Do they all share this session?

Reply:
Yes. All users share a single service account session.